### PR TITLE
Converted the AppModal to a AppPage.

### PR DESCRIPTION
**Pull request by aldipower**
Introducing PageComponent.
Moved components to be rendered into own function.
getInfo function with better default case.
Remove black background from page header.
Implement breadcrumbs for app and tasks.
Removed controls while displaying task details.
Darken the navbar background.
Breadcrumbs with darker background.
Adding hasHealth method to app model.

Rebased version of #1135 

@mlunoe

### CODE REVIEW

**Comment by aldipower** [comment]
m1
m2

File: src/main/resources/assets/js/components/AppPageComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    return {
+      activeViewIndex: 0,
+      activeTabId: tabs[0].id,
+      selectedTasks: {}
+    };
+  },
+
+  destroy: function () {
+    this.refs.pageComponent.destroy();
+  },
```
Comment (mlunoe) [inline src/main/resources/assets/js/components/AppPageComponent.jsx]: Why is this necessary?

File: src/main/resources/assets/js/components/AppBreadcrumbsComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    var taskClassSet = React.addons.classSet({
+      "active": true,
+      "hidden": activeViewIndex === 0
+    });
+
+    /* jshint trailing:false, quotmark:false, newcap:false */
+    /* jscs:disable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
+    return (
+      <ol className="breadcrumb">
+        <li key="apps"><a href="#" onClick={this.props.onDestroy}>Apps</a></li>
```
Comment (mlunoe) [inline src/main/resources/assets/js/components/AppBreadcrumbsComponent.jsx]: Let's get this down on a newline, so it alignes with the others:
```
<li key="apps">
  <a href="#" onClick={this.props.onDestroy}>Apps</a>
</li>
```

File: src/main/resources/assets/js/components/AppPageComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    if (!allTasksSelected) {
+      modelTasks.forEach(function (task) {
+        newSelectedTasks[task.id] = true;
+      });
+    }
+
+    this.setState({selectedTasks: newSelectedTasks});
+  },
+
+  toggleTask: function (task, value) {
```
Comment (mlunoe) [inline src/main/resources/assets/js/components/AppPageComponent.jsx]: I don't see this getting used anywhere? What is this for?

File: src/main/resources/assets/js/components/AppPageComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+          activeTask={this.props.activeTask}
+          activeViewIndex={this.state.activeViewIndex}
+          model={model}
+          onDestroy={this.props.onDestroy}
+          showTaskList={this.showTaskList} />
+        <div className="page-header">
+          <span className="h3 modal-title">{model.get("id")}</span>
+          <ul className="list-inline list-inline-subtext">
+            <li>
+                <span className={statusClassSet}>
```
Comment (mlunoe) [inline src/main/resources/assets/js/components/AppPageComponent.jsx]: Alignment here is off by one

Comment (mlunoe) [inline src/main/resources/assets/js/components/AppPageComponent.jsx]: Though I see that `TaskListComponent` has the function and, I guess, should be using it

File: src/main/resources/assets/js/components/AppPageComponent.jsx (hunk 3 above)
Comment (mlunoe) [inline src/main/resources/assets/js/components/AppPageComponent.jsx]: Though I see that TaskListComponent has the function and, I guess, should be using it
